# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/thunderbird/hotfix-sparc-tailcalls.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

error: cannot tail-call: target is not able to optimize the call into a sibling call
 5:52.28  1583 |             MUSTTAIL return fn(params, program, r,g,b,a);                                  \
 5:52.28       |                             ~~^~~~~~~~~~~~~~~~~~~~~~~~~~

--- firefox-142.0/gfx/skia/skia/include/private/base/SkFeatures.h.vanilla	2025-08-24 13:05:14.868190104 +0200
+++ firefox-142.0/gfx/skia/skia/include/private/base/SkFeatures.h	2025-08-24 13:05:55.768935250 +0200
@@ -73,6 +73,10 @@
   #define SK_CPU_PPC 1
 #endif
 
+#if defined(__sparc__)
+  #define SK_CPU_SPARC 1
+#endif
+
 /**
  *  SK_CPU_SSE_LEVEL
  *
--- firefox-142.0/gfx/skia/skia/src/core/SkRasterPipeline.h.vanilla	2025-08-24 13:03:51.224464329 +0200
+++ firefox-142.0/gfx/skia/skia/src/core/SkRasterPipeline.h	2025-08-24 13:08:58.894816219 +0200
@@ -27,7 +27,7 @@
 struct skcms_TransferFunction;
 
 #if __has_cpp_attribute(clang::musttail) && !defined(__EMSCRIPTEN__) && !defined(SK_CPU_ARM32) && \
-        !defined(SK_CPU_LOONGARCH) && !defined(SK_CPU_PPC) && \
+        !defined(SK_CPU_LOONGARCH) && !defined(SK_CPU_PPC) && !defined(SK_CPU_SPARC) && \
         !(defined(_WIN32) && defined(SK_BUILD_FOR_ANDROID_FRAMEWORK))
     // [[clang::musttail]] is disabled for the Android version of Skia running on Windows as it
     // causes crashes (This is probably related to http://crbug.com/1505442).
--- firefox-142.0/toolkit/components/protobuf/src/google/protobuf/port_def.inc.vanilla	2025-08-24 13:14:35.492924903 +0200
+++ firefox-142.0/toolkit/components/protobuf/src/google/protobuf/port_def.inc	2025-08-24 13:18:19.036124582 +0200
@@ -255,7 +255,7 @@
 #error PROTOBUF_TAILCALL was previously defined
 #endif
 #if __has_cpp_attribute(clang::musttail) && !defined(__arm__) && \
-    !defined(_ARCH_PPC) && !defined(__wasm__) &&                 \
+    !defined(_ARCH_PPC) && !defined(__sparc__) && !defined(__wasm__) &&                 \
     !(defined(_MSC_VER) && defined(_M_IX86)) &&                  \
     !(defined(__NDK_MAJOR__) && __NDK_MAJOR <= 24)
 #  ifndef PROTO2_OPENSOURCE
